DeliveryApi/cdk/src/main/java/com/ilmlf/delivery/db/DbStack.java

```diff
@@ -0,0 +1,249 @@
+import software.amazon.awscdk.services.secretsmanager.ISecret;
+import software.amazon.awscdk.services.secretsmanager.Secret;
+import software.amazon.awscdk.services.secretsmanager.SecretProps;
+import software.amazon.awscdk.services.secretsmanager.SecretStringGenerator;
+
+/**
+ * Database stack contains the followings.
+ * 1. Network resources (VPC, subnets, and Security Group)
+ * 2. RDS Database
+ * 3. RDS Proxy for connecting to database
+ * 4. Secrets for Lambda to connect to RDS Proxy
+ */
+@Getter
+public class DbStack extends Stack {
+  /**
+   * Database username for DB Proxy connection (via IAM Authorization).
+   */
+  private final String user;
+  private final String instanceEndpoint;
+  private final String proxyEndpoint;
+  private final Vpc vpc;
+  private final SecurityGroup securityGroup;
+  private final String proxyArn;
+
+  /**
+   * Contains database info, admin username, and password. This is a secret generated when
+   * creating the DB. Example of keys are:
+   * - username (admin username)
+   * - password (admin password)
+   * - engine (mysql
+   * - port (3306)
+   * - dbname (FarmerDB)
+   * - host (used for connection)
+   */
+  private final ISecret adminSecret;
+
+  /**
+   * Secret to beused by Lambda to connect to RDS Proxy via IAM authentication.
+   * - username (hardcoded to lambda_iam)
+   * - password
+   */
+  private final Secret userSecret;
+
+  /**
+   * Create a Database stack.
+   *
+   * @param scope used by superclass.
+   * @param id used by superclass.
+   * @param props used by superclass.
+   */
+  public DbStack(final Construct scope, final String id, final StackProps props) {
+    super(scope, id, props);
+
+    this.user = "lambda_iam";
+    String region = Stack.of(this).getRegion();
+
+    /**
+     * #################
+     * Network resources
+     * #################
+     *
+     * Create a VPC (Virtual Private Cloud), used for network partitioning.
+     *
+     * The VPC contains multiple "Subnets" that could be either Internet-public or private.
+     * Each Subnet is placed in different AZ (Availability Zones). Each AZ is in a different location
+     * within the region. In production, you should place your database and its replica in multiple AZ
+     * in case of failover. By default this stack deploys a database instance and its replica to different AZs.
+     */
+
+    // The `Vpc` construct creates subnets for you automatically
+    // See https://docs.aws.amazon.com/cdk/api/latest/docs/aws-ec2-readme.html#vpc for details
+    this.vpc = new Vpc(this, "farmer-Vpc");
+
+    // Security group acts as a virtual firewall to control inbound and outbound traffic
+    this.securityGroup =
+        new SecurityGroup(
+            this,
+            "FarmerDeliverySG",
+            SecurityGroupProps.builder()
+                .vpc(vpc)
+                .description("Shared SG for database and proxy")
+                .allowAllOutbound(true)
```

Comment:
Really required?

Reply:
outbound ports are usually left alone to default to *

Reply:
OK.